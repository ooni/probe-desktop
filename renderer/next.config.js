// const withSourceMaps = require('@zeit/next-source-maps')
// const withBundleAnalyzer = require('@next/bundle-analyzer')({
//   enabled: process.env.ANALYZE === 'true'
// })

// module.exports = {
//   // node: {global: true},
//   webpack: (config, options) => {
//     config.target = 'electron-renderer'

//     config.node = {
//       __dirname: false,
//       __filename: false,
//       global: true
//     }

//     // config.plugins.push(
//     //   new options.webpack.DefinePlugin({
//     //     'process.env.NEXT_IS_SERVER': JSON.stringify(
//     //       options.isServer.toString()
//     //     )
//     //   })
//     // )

//     // config.module.rules.push({
//     //   test: /\.(eot|ttf|woff|woff2|otf)$/,
//     //   use: [
//     //     {
//     //       loader: 'url-loader'
//     //     }
//     //   ]
//     // })
//     // if (!options.isServer) {
//     //   config.resolve.alias['@sentry/node'] = '@sentry/browser'
//     // }
//     return config
//   },
// }

const nextConfig = {
  output: 'export',
  webpack: (config) => {
    config.module.rules.push({
      test: /\.svg$/,
      use: ['@svgr/webpack'],
    });
    return config;
  },
}

module.exports = nextConfig
