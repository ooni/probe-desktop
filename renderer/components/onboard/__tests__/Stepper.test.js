import React from 'react'
import { theme } from 'ooni-components'
import { ThemeProvider } from 'styled-components'
import renderer from 'react-test-renderer'

import Stepper from '../Stepper'

describe('Tests from Stepper component', () => {
  test('With active index 0', async () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Stepper activeIdx={0} />
        </ThemeProvider>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  test('With active index 1', async () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Stepper activeIdx={1} />
        </ThemeProvider>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
  
  test('With active index 2', async () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Stepper activeIdx={2} />
        </ThemeProvider>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
