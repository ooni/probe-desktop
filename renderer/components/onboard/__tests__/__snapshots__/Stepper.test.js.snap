// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests from Stepper component With active index 0 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  justify-content: center;
  align-items: center;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  display: flex;
}

.c3 {
  background-color: #f8f9fa;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c5 {
  background-color: #0588cb;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c4 {
  background-color: #0588cb;
  width: 50px;
  height: 3px;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
  />
  <div
    className="c2 c4"
  />
  <div
    className="c2 c5"
  />
  <div
    className="c2 c4"
  />
  <div
    className="c2 c5"
  />
  <div
    className="c2 c4"
  />
  <div
    className="c2 c5"
  />
</div>
`;

exports[`Tests from Stepper component With active index 1 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  justify-content: center;
  align-items: center;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  display: flex;
}

.c3 {
  background-color: #f8f9fa;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c6 {
  background-color: #0588cb;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c4 {
  background-color: #f8f9fa;
  width: 50px;
  height: 3px;
}

.c5 {
  background-color: #0588cb;
  width: 50px;
  height: 3px;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
  />
  <div
    className="c2 c4"
  />
  <div
    className="c2 c3"
  />
  <div
    className="c2 c5"
  />
  <div
    className="c2 c6"
  />
  <div
    className="c2 c5"
  />
  <div
    className="c2 c6"
  />
</div>
`;

exports[`Tests from Stepper component With active index 2 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  justify-content: center;
  align-items: center;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c1 {
  display: flex;
}

.c3 {
  background-color: #f8f9fa;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c6 {
  background-color: #0588cb;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c4 {
  background-color: #f8f9fa;
  width: 50px;
  height: 3px;
}

.c5 {
  background-color: #0588cb;
  width: 50px;
  height: 3px;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
  />
  <div
    className="c2 c4"
  />
  <div
    className="c2 c3"
  />
  <div
    className="c2 c4"
  />
  <div
    className="c2 c3"
  />
  <div
    className="c2 c5"
  />
  <div
    className="c2 c6"
  />
</div>
`;
